output.go

```diff
@@ -0,0 +1,212 @@
+package timescaledb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/sirupsen/logrus"
+
+	"go.k6.io/k6/output"
+	"go.k6.io/k6/stats"
+)
+
+func init() {
+	output.RegisterExtension("timescaledb", newOutput)
+}
+
+func newOutput(params output.Params) (output.Output, error) {
+	config, err := getEnvConfig(params.Environment)
```

Comment:
I still don't entirely understand the source of thresholds in params or the url that I should parse in `getConsolidatedConfig`:
* for `thresholds` , should it simply be in the js k6 script? because in the original WIP PR, they seemed to be parsed as `submetrics`--is there a difference? (I don't really understand the big picture of k6 yet)
* for the `url` is it a command line argument? I couldn't find if there was a specific flag for it; it almost looked like it was the second argument, but that doesn't really make sense.

Reply:
As I mention in https://github.com/grafana/xk6-output-timescaledb/pull/1#discussion_r682387840 you should get the thresholds through the API that will give the already parsed thresholds to any output instead of preparing them from the Options. I really don't want to go into the details of thresholds, metrics and submetrics at this time ... especially as I hope we will get to rewriting them soon :tm: 

The `output.Params` are generated [here](https://github.com/grafana/k6/blob/a733b306b344077ab576d2ff9a07d3c6d1a16173/cmd/outputs.go#L102-L123) and the `url`/`params.ConfigArgument` is the part after the equals `=` sign if you run it like `k6 run -o timescaledb=here-this-is-the-url/config-argument script.js`. It is just very common (in my experience at least) to do things like that across different tools and we have picked it up. It works less great if the actual argument needs to be a *real* URL as you then have arguments that need to be parsed and set as config and possibly some that need to stay in the URL.  Sometimes (as the name in `output.Params` eludes) it is not a URL it's just an additional param like in the case of the CSV output you can provide a filename to which to write the CSV instead of to the screen.

And apparently, Robin decided that it's a good idea to provide configuration as you can notice in the original docker-compose.yml change `K6_OUT=timescaledb=postgresql://timescaledb:5432/k6`, which looks very natural to me as well

Reply:
The config argument at least when I was testing locally seemed to be the entire part after the `=` (`here-this-is-the-url/config-argument` in your example). That is what you meant?

In the original collector WIP, the full url (config argument) was passed to be parsed by pgx, but I've had to take apart any query arguments from the full url so that only the first part of the url is passed to pgx.ParseConfig (which won't take any invalid query parameter).

Default values were left as in the original collector WIP. 

Let me know what you think.

Reply:
> The config argument at least when I was testing locally seemed to be the entire part after the = (here-this-is-the-url/config-argument in your example). That is what you meant?

yes. Out of curiosity, how did you understand it given, my explanation? Just so I know what I did wrong.

I am pretty sure that only the part without timescaledb= was provided even then ... but I might be wrong :shrug:

Reply:
Just at first I thought you meant there was another params.URL (::shrug::) and params.ConfigArgument was everything in the path. My bad!

(I believe) The original commit included `c.URL = null.NewString(u.String(), true)` which would assign the full url to c.URL and later it was passed as `connParams, err := pgx.ParseURI(conf.URL.String)` (Not that it matters for the first part... the parseURL function was not even called...it was using the default value)

I remember this returning errors when I was playing around in earlier versions of my commit.

I'm going to resolve the conversation, but feel free to re-open a conversation if there is another issue.